File: XMLCheck/Handler/CheckRulesXML07/Xml7RuleRunner.cs
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML07
{
    public static class Xml7RuleRunner
    {
        private static readonly List<IXml7Rule> _rules = new()
        {

        };
        public static List<ErrorDetails> Run(Xml7Model? model,HoSoContext context)
        {
            var errors = new List<ErrorDetails>();
            foreach (var rule in _rules)
            {
                var err = rule.Check(model,context);
                if (err != null)
                    errors.Add(err);
            }

            return errors;
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml7/xml7RuleRunner.ts
import { Xml7Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';
import { IXml7Rule } from './xml7RuleBase';

export class Xml7RuleRunner {
  private static readonly rules: IXml7Rule[] = [];

  public static run(model: Xml7Model, context: HoSoContext): ErrorDetails[] {
    const errors: ErrorDetails[] = [];
    if (!model) return errors;

    for (const rule of this.rules) {
      const err = rule.check(model, context);
      if (err) {
        errors.push(err);
      }
    }

    return errors;
  }
}
